prov/psm/src/psmx.h

```diff
@@ -451,7 +451,10 @@ struct psmx_trigger {
 };
 
 struct psmx_fid_cntr {
-	struct fid_cntr		cntr;
+	union {
+		struct fid_cntr		cntr;
+		struct util_cntr	util_cntr; /* for util_poll_run */
+	};
```

Comment:
Couldn't you replace fid_cntr with util_cntr instead?

On that note, does psmx_fid_domain include the util_domain structure?